ci_framework/roles/libvirt_manager/tasks/deploy_layout.yml

```diff
@@ -56,23 +65,51 @@
       --ssh-inject {{ _admin_user }}:file:{{ ansible_user_dir }}/.ssh/authorized_keys
       | tee -a {{ item.value.image_local_dir }}/{{ item.value.disk_file_name }}.log
     creates: "{{ item.value.image_local_dir }}/{{ item.value.disk_file_name }}.log"
-  loop: "{{ cifmw_libvirt_manager_configuration.vms | dict2items }}"
+  loop: "{{ _layout.vms | dict2items }}"
   loop_control:
     label: "{{ item.key }}"
 
+- name: Create temporary ssh keypair
+  ansible.builtin.command:
+    cmd: >-
+      ssh-keygen -t ed25519 -f .ssh/cifmw_reproducer_key -P '' -q
+    creates: .ssh/cifmw_reproducer_key
+
+- name: Slurp public key for later use
+  register: pub_ssh_key
+  ansible.builtin.slurp:
+    path: .ssh/cifmw_reproducer_key.pub
+
+- name: Slurp private key for later use
+  register: priv_ssh_key
+  ansible.builtin.slurp:
+    path: .ssh/cifmw_reproducer_key
+
+- name: Create fact holding network data for VMs
+  ansible.builtin.set_fact:
+    cacheable: true
+    cifmw_reproducer_network_data: {}
```

Comment:
Where is this one used? In another PR?

Reply:
in the reproducer, yeah. It's kind of inter-dependent